client/src/components/menu-bar/MenuBar.js

```diff
@@ -0,0 +1,107 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import AppBar from '@material-ui/core/AppBar';
+import IconButton from '@material-ui/core/IconButton';
+import Typography from '@material-ui/core/Typography';
+import MenuItem from '@material-ui/core/MenuItem';
+import Menu from '@material-ui/core/Menu';
+import Toolbar from '@material-ui/core/Toolbar';
+import AccountCircle from '@material-ui/icons/AccountCircle';
+import Avatar from '@material-ui/core/Avatar';
+import auth from '../../auth/auth';
+import config from '../../config/globalConfig';
+
+export default class MenuBar extends Component {
+  state = {
+    loggedIn: false,
+    user: {},
+    anchorEl: null,
+  };
+
+  componentDidMount() {
+    this.fetchUserIfLoggedIn();
+  }
+
+  componentDidUpdate() {
+    this.fetchUserIfLoggedIn();
+  }
+
+  fetchUserIfLoggedIn = () => {
+    const { loggedIn } = this.state;
+    if (!loggedIn && auth.isAuthenticated()) {
+      auth.fetchUserData()
+        .then((data) => {
+          const { user = {} } = data;
+          this.setState({ loggedIn: true, user });
+        })
+        .catch(() => {
+          this.setState({ loggedIn: false, user: {} });
+        });
+    }
+  }
+
+  handleMenu = (event) => {
+    this.setState({ anchorEl: event.currentTarget });
+  }
+
+  handleClose = () => {
+    this.setState({ anchorEl: null });
+  }
+
+  handleLogout = () => {
+    const { history } = this.props;
+    this.handleClose();
+    auth.logout(() => {
+      this.setState({ loggedIn: false, user: {} });
+      history.push(config.loginRoute);
+    });
+  }
+
+  renderUserAvatar = (user) => {
```

Comment:
pass picture directly instead of user. it does not use anything else from user.

Reply:
Changed